integration/spark/src/test/java/openlineage/spark/agent/SparkContainerIntegrationTest.java

```diff
@@ -0,0 +1,215 @@
+package openlineage.spark.agent;
+
+import static java.nio.file.Files.readAllBytes;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.JsonBody.json;
+
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.matchers.MatchType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.MockServerContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@Tag("integration-test")
+@Testcontainers
+public class SparkContainerIntegrationTest {
+
+  private static final Network network = Network.newNetwork();
+  @Container private static final PostgreSQLContainer<?> postgres = makePostgresContainer();
+
+  @Container
+  private static final MockServerContainer openLineageClientMockContainer =
+      makeMockServerContainer();
+
+  private static GenericContainer<?> pyspark;
+  private static MockServerClient mockServerClient;
+
+  @BeforeAll
+  public static void setup() {
+    mockServerClient =
+        new MockServerClient(
+            openLineageClientMockContainer.getHost(),
+            openLineageClientMockContainer.getServerPort());
+    mockServerClient
+        .when(request("/api/v1/lineage"))
+        .respond(org.mockserver.model.HttpResponse.response().withStatusCode(200));
```

Comment:
I think it ought to be a `201`, right?

Reply:
You are correct should be 201, we  have POST there, thanks